osu.Game.Rulesets.Osu/Skinning/LegacyMainCirclePiece.cs

```diff
@@ -98,20 +101,36 @@ Texture getTextureWithFallback(string name)
             }
         }
 
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            state.BindValueChanged(updateState, true);
+            accentColour.BindValueChanged(colour => hitCircleSprite.Colour = colour.NewValue, true);
+            indexInCurrentCombo.BindValueChanged(index => hitCircleText.Text = (index.NewValue + 1).ToString(), true);
+        }
+
         private void updateState(ValueChangedEvent<ArmedState> state)
         {
             const double legacy_fade_duration = 240;
 
             switch (state.NewValue)
             {
                 case ArmedState.Hit:
-                    this.FadeOut(legacy_fade_duration, Easing.Out);
-
-                    hitCircleSprite.ScaleTo(1.4f, legacy_fade_duration, Easing.Out);
-                    hitCircleOverlay.ScaleTo(1.4f, legacy_fade_duration, Easing.Out);
-
-                    if (skin.GetConfig<OsuSkinConfiguration, bool>(OsuSkinConfiguration.ExpandNumberPiece)?.Value ?? true)
+                    circlePieces.FadeOut(legacy_fade_duration, Easing.Out);
+                    circlePieces.ScaleTo(1.4f, legacy_fade_duration, Easing.Out);
+
+                    var legacyVersion = skin.GetConfig<LegacySetting, decimal>(LegacySetting.Version)?.Value;
+
+                    if (legacyVersion >= 2.0m)
+                        // legacy skins of version 2.0 and newer apply immediate fade out to the number piece and only that.
+                        hitCircleText.FadeOut(legacy_fade_duration / 4, Easing.Out);
```

Comment:
I'm really confused by this comment. It states that "newer skins apply immediate fade out to the number piece and only that", which is immediately followed by a fade of the text with a non-zero (albeit shorter) duration. I'm guessing you meant "scale"?

Reply:
not sure what you mean by "scale", immediate fade out as in the duration is extremely short (mostly 2 frames of transforming) if that's what you're confused about?

Reply:
Both the terminology (not sure what a "number piece" is supposed to mean in this context) and the non-immediate immediate fade out (for me immediate is duration 0 always; else this reads like a bug - what you have there is more like a very short fade consistent with stable).

Reply:
> Both the terminology (not sure what a "number piece" is supposed to mean in this context)

I'm not sure if there's a problem with saying "number piece" in a `LegacyMainCirclePiece` component referring to the hitcircle number sprite text, it's quite clear to me.

> and the non-immediate immediate fade out (for me immediate is duration 0 always; else this reads like a bug - what you have there is more like a very short fade consistent with stable).

Ah, if there's one thing I'm missing in my contributions, it's proper English; will go with `very short fade out`.